### DISCUSSION TITLE
v2: Transformer + Namer + Packager

### DISCUSSION THREAD

**Discussion by eirikb** [parcel-bundler/parcel#6086] [Q&A]
Hi

How to use Packager in a general way when Namer changes the name?  
Is there some other way to trigger the Packager beside file name?

I'm writing a mustache plugin and wanted to create a general **parcel-config** file users could use in their _.parcelrc_-file.  
Plan is: 
hello.mustache.txt ➡️ mustache transform ➡️ rename to hello.txt ➡️ replace URL hashes with file names ➡️ dist/hello.txt

Current version picks up `*.mustache.*`-files:
```json
{
  "namers": [
    "@eirikb/parcel-namer-mustache",
    "@parcel/namer-default"
  ],
  "transformers": {
    "*.mustache.*": [
      "@eirikb/parcel-transformer-mustache"
    ]
  },
  "packagers": {
    "*.mustache.*": "@parcel/packager-raw-url"
  }
}
```
Problem here is that the packager won't hit, because file has been renamed by `parcel-namer-mustache` to not include the ".mustache."-part. 
I also tried using .mustache, but never managed to have the namer remove an extension, and it would not help the packager anyway.

**Comment by mischnic** [comment]
There are two ways to change the type:
1. in the transformer by changing `asset.type`
2. in a packager or optimizer, you can return a `type` that replaces the previous extension: https://github.com/parcel-bundler/parcel/blob/ff1a55e8a8d0ec989adfede698771b7fdb567d95/packages/core/types/index.js#L1077-L1082

---

Changing the type to `mustache` in the transformer and then setting it to `txt` in the packager might work for you (so that you can do `"packagers": { "*.mustache"`)

**Reply by eirikb** [reply]
I'm not able to make it work.
I've tried this:

1. Entry: hello.mustache.txt
2. Transformer: Run mustache. Set `asset.type = 'mustache'`
3. Namer: Replace ".mustache" with ""  
Fails:
> parcel-namer-mustache: Destination name hello.txt extension does not match bundle type "mustache"

Another try:

1. Entry: hello.txt.mustache
2. Transformer: Run mustache
3. Namer: Replace ".mustache" with ""
Fails:
> parcel-namer-mustache: Destination name hello.txt extension does not match bundle type "mustache"

Another try:

1. Entry: hello.txt.mustache
2. Transformer: Run mustache
3. Namer: Do nothing
4. Packager: `return { content, type: 'txt' }`
5. dist: **hello.txt.mustache**
